v2/protocol/http/utility.go

```diff
@@ -0,0 +1,31 @@
+/*
+ Copyright 2022 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package http
+
+import (
+	"context"
+	nethttp "net/http"
+
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/event"
+)
+
+// NewEventFromHttpRequest returns an Event.
+func NewEventFromHttpRequest(req *nethttp.Request) (*event.Event, error) {
+	msg := NewMessageFromHttpRequest(req)
+	return binding.ToEvent(context.Background(), msg, nil)
```

Comment:
Found the issue,

```suggestion
	return binding.ToEvent(context.Background(), msg)
```

Because `ToEvent` is defined as `ToEvent(ctx context.Context, message MessageReader, transformers ...Transformer) (*event.Event, error)`, the iterator is attempting to call a nil transformer, note that transformers is a vararg and nil is a valid inner slice member. So when the code hits this:

```go
	for _, transformer := range t {
		err := transformer.Transform(r, w)
...
```

`transformer` is nil, because `t` has been created eq to `[]Transformer{nil}`, which is a slice of len 1, and `t[0]` happens to be `nil`.


To solve this, don't pass `nil` to the `vararg` (`, t ...Transformer`) param.